### PR TITLE
Add mechanism to control non-determinism in retries

### LINKED ISSUE #155: Add mechanism to control non-determinism in retries
**What would you like to be added**:
On the [retry definition](https://github.com/serverlessworkflow/specification/blob/master/specification.md#retry-definition) there is an option to increase the time period to be waited between attempts ("multiplier"). Many exponential backoff implementations also make use of a (usually small) bounded random amount of time that is added to the delay.

It would be nice if this bounded random amount of time (sometimes referred to as "jitter") was part of the spec, it can be very important to certain workloads and IMO it's better that this is enforced by the specification than to be left up to the implementation.

This parameter could be of type integer and be measured in milliseconds. The delay for the `n-th` attempt could be defined as: `interval * exp(multiplier, n-1) + rand(0, jitter)`, though I believe the exact formula could be left up to the implementation. The key point here is enforcing a mechanism to provide bounded randomness.

**Why is this needed**:

This is important in scenarios where the transient failure is a result of a race condition (for example) between two actions which could be, by bad luck, causing the failure and waiting the same amount of time before retrying and causing the failure again, only to wait for the same amount of time again (in this scenario, their "interval" and "multiplier" would be the same).

### PR DESCRIPTION
**Pull request by LCaparelli**
Signed-off-by: Lucas Caparelli <lucas.caparelli112@gmail.com>

**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [X] Specification
- [X] Schema
- [ ] Examples
- [ ] Usecases
- [ ] Extensions
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:

Add mechanism to control non-determinism in retries. Fix #155 

**Special notes for reviewers**:

I was not sure if the spec and the schema is everything I needed to change. Sorry if other things needed modification as well, just let me know and I'll add them :-)

**Additional information (if needed):**

### CODE REVIEW

File: schema/workflow.json [hunk 1]
```diff
@@ -339,6 +339,10 @@
           "minimum": 0,
           "minLength": 0,
           "description": "Maximum number of retry attempts. Value of 0 means no retries are performed"
+        },
+        "jitter": {
+          "type": "string",
+          "description": "Maximum random amount of time added or subtracted from the delay between each retry (ISO 8601 repeatable format)"
```
Comment (tsurdilo) [inline schema/workflow.json:345]: should not be repeating intervals, but time duration format

File: specification.md [hunk 2]
```diff
@@ -1332,6 +1332,7 @@ transition:
 | interval | Interval value for retry (ISO 8601 repeatable format). For example: "R5/PT15M" (Starting from now repeat 5 times with 15 minute intervals)| string | no |
 | multiplier | Multiplier value by which interval increases during each attempt (ISO 8601 time format). For example: "PT3S" meaning the second attempt interval is increased by 3 seconds, the third interval by 6 seconds and so on | string | no |
 | maxAttempts | Maximum number of retry attempts. Value of 0 means no retries are performed | string or integer | no |
+| jitter | Maximum random amount of time added or subtracted from the delay between each retry (ISO 8601 repeatable format) | string | no |
```
Comment (tsurdilo) [inline specification.md:1335]: same comment here regarding "repeating", should be time duration

File: schema/workflow.json (hunk 1 above)
Comment (tsurdilo) [inline schema/workflow.json:345]: I wonder if we can reword this a little, saying this parameter represents a "random amount" might mean that its like a random number, different for each instance of this particular workflow, but it's not.

**Review by tsurdilo — CHANGES_REQUESTED** [review]

File: schema/workflow.json (hunk 1 above)
Comment (LCaparelli) [reply schema/workflow.json:345]: Well, the amount of time that gets added is the random part, so maybe we could move the "random" word a bit further down, like:

```suggestion
          "description": "Maximum amount of random time added or subtracted from the delay between each retry (ISO 8601 repeatable format)"
```

I think this way it's harder to interpret that the maximum amount is random. Wdyt?

Comment (tsurdilo) [reply schema/workflow.json:345]: +1

**Review by tsurdilo — APPROVED** [review]

**Comment by tsurdilo** [comment]
@LCaparelli looks good to me. thanks! We will go over it in our meet on Monday to get others opinions as well.

**Comment by tsurdilo** [comment]
@LCaparelli looks good to me w/ the changes discussed. We can merge after updates.

**Comment by LCaparelli** [comment]
@tsurdilo done, thanks for the review!

Should I be concerned with the failing action? Looks like the docker build is failing due to a failure in `npm`. Could the changes have caused that?

**Comment by tsurdilo** [comment]
@LCaparelli not sure why this "greeting" action is failing honestly. I'll just remove it.

**Comment by tsurdilo** [comment]
@LCaparelli removed that action. Mind amending this pr? For example just add your name to: https://github.com/serverlessworkflow/specification/blob/master/community/contributors.md

**Review by ricardozanini — APPROVED** [review]

**Comment by tsurdilo** [comment]
Waiting on @LCaparelli to update jitter param to be either string or integer value as discussed and then for @jorgenj to take a look at approve. Thanks guys!

**Comment by LCaparelli** [comment]
@tsurdilo @jorgenj done, please let me know your thoughts :-)

**Review by jorgenj — APPROVED** [review]
LGTM!

**Pull request merged**